### PR TITLE
added user attributes - previously missing

### PR DESCRIPTION
**Pull request by izabellemarianne**

### CODE REVIEW

File: src/main/java/db/DBUserDataAccessObject.java
Comment (ethanliu24) [inline src/main/java/db/DBUserDataAccessObject.java:1]: I've implemented this file to incorporate MongoDB (see #17), so you can remove this file from staging to avoid merge conflicts

File: src/main/java/entity/CommonUser.java
Comment (ethanliu24) [inline src/main/java/entity/CommonUser.java:31]: Pass `0` through the constructor and assign it to this.accessLevel (i.e. `this.accessLevel = accessLevel`). We'll create users through factories, so the initializations should go there.

**EDIT**: on a second thought, I think we should pass in 0 into UserFactory instead, so in the create method, it should take in a parameter called accessLevel or something

File: src/main/java/entity/CommonUserFactory.java
Comment (ethanliu24) [inline src/main/java/entity/CommonUserFactory.java]: As to the comment before, add parameter for accessLevel.

File: src/main/java/use_case/signup/SignupInteractor.java
Comment (ethanliu24) [inline src/main/java/use_case/signup/SignupInteractor.java:36]: For SignupOutputData, it should contain *all* the information signupInputData has (e.g. userID, password, email, etc). That said, update SignupOutputData to take the relavent parameters.

File: src/main/java/use_case/signup/SignupInteractor.java
Comment (ethanliu24) [inline src/main/java/use_case/signup/SignupInteractor.java]: Check the main review comment, we'll create an empty list here for moderating and posts attribute

File: src/main/java/entity/UserFactory.java
Comment (ethanliu24) [inline src/main/java/entity/UserFactory.java]: update the function parameters accordingly

**Review by ethanliu24 — CHANGES_REQUESTED** [review]
Nice job! Just some minor changes needed, there are two attributes missing for users:
- [ ] A `moderating` attribute to store a list of strings (representing posts ids) that the user is moderating
- [ ] A `posts` attribute to store a list of strings (representing post ids) of posts that the user created

Use ArrayList for both of these attributes, see my comments on where to pass them in. Also, remember to update the related classes/functions to these changes. 

One last thing, after I set up the database, I feel like accessLevel is not needed in our program anymore - we could just check who's the current user in the DAO and determine if they could do actions. However we could totally keep it in (I'll let u decide that!), but please let me know if you kept it in or not so I could update my DAO implementation accordingly.

Hopefully after you made these changes, we could merge it in and have our user DAO working!

**Comment by ethanliu24** [comment]
@izabellemarianne we only need the changes in the `entity` folder. Please remove everything else from this PR (i.e. unstage from git)

**Comment by izabellemarianne** [comment]
Sorry about that, I have pushed another commit does that work?

**Review by ethanliu24 — CHANGES_REQUESTED** [review]
Hey @izabellemarianne, could u also remove the files `src/main/java/app/ConnectHub.java`, `src/main/java/controller/ViewModel.java`, `src/main/java/signup/SignupController.java`, `src/main/java/db/DBUserDataAcessObject.java` as well? Right now in the `Files Changed` tab, you can see that these file are still there, but after u unstage or remove them, you should not see them in there any more.

**Pull request closed without merging**
